documentation/sphinx/source/bulkdump.rst

```diff
@@ -0,0 +1,102 @@
+##############################
+BulkDump (Dev)
+##############################
+
+| Author: Zhe Wang
+| Reviewer: Michael Stack, Jingyu Zhou
+| Audience: FDB developers, SREs and expert users.
+
+
+Overview
+========
+In a FoundationDB (FDB) key-value cluster, every key-value pair is replicated across multiple storage servers. 
+The BulkDump tool is developed to dump all key-value pairs within the input range to files.
+Note that when the input range is large, the range splits into smaller ranges.
+Each subrange data is dumped to a file at a version. All data within a file is at the same version. However, different file versions can be different.
+
+Input and output
+----------------
+When a user wants to start a bulkdump job, the user provides the range to dump and the path root to dump the data. 
+The range can be any subrange within the user key space (i.e. " " ~ "\\xff").
+Dumping the data of the system key space and special key space (i.e. "\\xff" ~ "\\xff\\xff\\xff") is not allowed.
+The path root can be either a blobstore url (TBD) or a path of a file system.
+Given the input range, if the range is large, the range splits into smaller ranges.
+Each subrange is dumped at a version to a folder. In particular, the folder is organized as following:
+
+1. (rootLocal)/(relativeFolder)/(dumpVersion)-manifest.txt (must have)
+2. (rootLocal)/(relativeFolder)/(dumpVersion)-data.sst (omitted if the subrange is empty)
+3. (rootLocal)/(relativeFolder)/(dumpVersion)-sample.sst (omitted if data size is too small to have a sample)
+
+The (relativeFolder) is defined as (JobId)/(TaskId)/(BatchId). 
+The (dumpVersion) is the version of the data stored in the (dumpVersion)-data.sst file.
+At any time, a FDB cluster can have at most one bulkdump job. 
+A bulkdump job is partitioned into tasks by range and aligned to the shard boundary.
+When dumping the range of a task, the data is collected in batches. All key-value pairs of a batch are collected at the same version.
+Above all, (JobId) is the unique ID of a job. (TaskId) is the unique ID of a task. (BatchId) is the unique ID of a batch.
+All tasks's data files of the same job locates at the same Job folder named by the JobId.
```

Comment:
I'm a bit confused on BatchID. Is it the dumpVersion? Or is a batch a number of files that all have the same dumpVersion? (But it seems like each file gets a unique version else we'd overwrite files with matching versions?).